Zend/zend_alloc.c

```diff
@@ -245,6 +248,7 @@ struct _zend_mm_heap {
 	size_t             size;                    /* current memory usage */
 	size_t             peak;                    /* peak memory usage */
 #endif
+	uintptr_t          key;                     /* free slot shadow ptr key */
```

Comment:
```suggestion
	uintptr_t          key;                     /* free slot shadow ptr xor key */
```

To explain its type: it needs to be a `uintptr_t` since it's xored with pointers. Also, the name `shadow ptr` is a bit confusing, since it's not really a pointer it self, but a key to encrypt the `free slot ptr`.